[spatuly] iP

spatuly opened the pull request:
## UkePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

UkePro remembers things you need to do for you. It's,
- text-based
- easy to learn
- ~~FAST~~ _VERY_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. run it
3. add your tasks
4. leave it to remember your tasks for you! 😄 

And it is **FREE**! Get it **NOW**!

Features:
- [x] Managing tasks (with and without deadlines)
- [x] Search for added tasks
- [ ] \(Coming soon) Reminders 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public static void main(String[] args) {
    new Duke(FILEPATH).run();
}
```

spatuly committed:
Level-0

spatuly committed:
Level-0 v2 New Name

spatuly committed:
Level-1

spatuly committed:
Level-2

spatuly committed:
Level-3

spatuly committed:
Level-4

spatuly committed:
A-TextUiTesting

spatuly committed:
Level-5

spatuly committed:
Level-6

spatuly committed:
Implement saving and loading tasks

spatuly committed:
Merge branch 'branch-Level-7'

spatuly committed:
Implement LocalDateTime in Deadline and Event tasks

spatuly committed:
Merge branch 'branch-Level-8'

spatuly committed:
Integrate OOP principles

spatuly committed:
Put all classes into one package

spatuly committed:
Add gradle

spatuly committed:
Implement JUnit tests for Todo and Deadline methods

spatuly committed:
Add JavaDoc for all classes

spatuly committed:
Merge branch 'branch-A-JavaDoc'

spatuly committed:
Fix bug with exception catching

spatuly committed:
Update codebase to adhere to coding standards

spatuly committed:
Merge branch 'branch-A-CodingStandard'

spatuly committed:
Implement find command

spatuly committed:
Merge branch 'branch-Level-9'

spatuly committed:
Remove data directory and text file

spatuly committed:
Add checkStyle and update codebase to pass checkStyle

kiwibang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+     * @param storedTasks TaskList to be printed.
+     * @param isListCommand Whether the method is called from a list command.
+     */
+    public void printList(TaskList storedTasks, boolean isListCommand) {
```
Perhaps can avoid nesting by having a variable to conditionally store different strings depending on the value of isListCommand? 🤔 
For example: 

```java
String initialMessageToUser = isListCommand
        ? "Okay! Here's the full list of your added tasks:"
        :  "Here's the list of matching tasks I found:";
System.out.println(intialMessageToUser);
```

kiwibang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+        int len = storedTasks.getLength();
+
+        if (isListCommand) {
+            if (len > 0) {
```
Perhaps can add this as a guard clause at the start of the method? Since the function can terminate if the list is empty 🤔 
For example:

```java
if (len == 0) {
    System.out.println("No tasks found!");
    return; // terminates the method early since there is nothing to print
}
```

kiwibang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+                    System.out.println(i + ". " + storedTasks.getTask(i - 1));
+                }
+            } else {
+                System.out.println("No matching tasks found :(");
```
Perhaps can add another String variable to store a string value conditionally if this error message needs to be printed out? 🤔 
For example:
```java
String emptyTasksMessage = isListCommand
        ? "No tasks found!"
        : "No matching tasks found :(";
```

kiwibang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,271 @@
+            ui.printError(e);
+        }
+
+        while (hasNext) {
```
Perhaps this can be shortened to multiple methods? 🤔

kiwibang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,271 @@
+        try {
+            tasks = new TaskList(storage.loadTaskList());
+        } catch (DukeException e) {
+            ui.printError(e);
```
Perhaps can avoid using try-catch as a control flow statement? I believe there is a method to check if the database file exists using Java IO classes 🤔

kiwibang reviewed (commented):
LGTM! Just some nits to fix. Better than my PR honestly. 👍

thienmy0 commented on src/main/java/duke/DukeDataFileException.java:
```diff
@@ -0,0 +1,16 @@
+ */
+public class DukeDataFileException extends DukeException {
+
+    /**
+     * Returns the string representation of the exception.
+     */
+    @Override
+    public String toString() {
```
Do overriden methods need javadoc? Else is there a return for this method that is not mentioned in the javadoc?

thienmy0 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,271 @@
+    /**
+     * Generates list of added tasks.
+     */
+    public void handleList() {
```
Not sure I like the naming of this method since it doesn't give me the impression that it is meant to generate the list

thienmy0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+        return input.nextLine();
+    }
+
+    /**
+     * Greets the user when the chatbot is started.
+     */
```
I like how all your Javadoc comments are very clear and straight to the point!

thienmy0 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,271 @@
+ */
+public class Duke {
+
+    private static final String FILEPATH = "./data/duke.txt";
+    private static boolean hasNext = true;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
```
Good job on using the appropriate naming convention for variables throughout your code :+1:

thienmy0 reviewed (commented):
Your code looks neat to me!
I honestly struggled to find issues.
Keep up the good work :smile:

spatuly committed:
Implement GUI for Uke

spatuly committed:
Merge branch 'branch-Level-10'

spatuly committed:
Improve printList in Ui.java

Certain parts of the code has unnecessary deep nesting and repetition.
Guard clause is also absent.

Editing the code improves code quality.

Let's remove repetition by adding variables to store different strings
conditionally, and add a guard clause.

These avoid unnecessary nesting and allow for early termination.

spatuly committed:
Add assertions

spatuly committed:
Merge pull request #1 from spatuly/branch-A-Assertions

Add assertions

spatuly committed:
Merge pull request #2 from spatuly/branch-A-CodeQuality

Improve printList in Ui.java

spatuly committed:
Merge branch 'master' into branch-A-CodeQuality

spatuly committed:
Merge pull request #3 from spatuly/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

spatuly committed:
Implement view command

spatuly committed:
Merge branch 'branch-B-ViewSchedules'

spatuly committed:
Give Uke more personality

spatuly committed:
Change FXML to display full list with no ellipsis

spatuly committed:
Improve GUI of chatbot

spatuly committed:
Add files via upload

spatuly committed:
Update README.md

spatuly committed:
Update README.md

spatuly committed:
Update README.md

spatuly committed:
Update README.md

spatuly committed:
Update JUnit tests

spatuly committed:
Merge branch 'master' of https://github.com/spatuly/ip

spatuly committed:
Add Javadocs for new methods

spatuly committed:
Remove redundant import statements

spatuly committed:
Update codebase to pass checkStyle
